src/hotspot/share/gc/shared/genCollectedHeap.cpp

```diff
@@ -132,6 +133,8 @@ jint GenCollectedHeap::initialize() {
 
   GCInitLogger::print();
 
+  GCForwarding::initialize(_reserved, SpaceAlignment);
```

Comment:
I think SpaceAlignment is correct. We want to pass a region-size there, and the (default) region size for Serial should be the space alignment, because that is what eden, survivors and old-space will be aligned at. Unfortunately, Serial GC doesn't generally slide from top to bottom: it starts to slide old into old, then young into old until old is full, then slide the rest into young. Even worse, the survivor spaces are swapped with every GC cycle, so we really don't know that sliding goes top -> bottom. Using 'virtual' regions that align at SpaceAlignment solves the problem, though.
(One exception is when the whole heap fits into our 2^28 words range, in which case we can treat the whole heap as single region)
That said, I see a bug in the line: GCForwarding::initialize() takes region size *in words* but SpaceAlignment is *in bytes*. I'm fixing that to passing space-alignment in words instead.

Reply:
Ah, that is _region size_, okay. `SpaceAlignment` seems okay then.